src/Plugin/views/display/DataExport.php

```diff
@@ -28,6 +31,38 @@ class DataExport extends RestExport {
   /**
    * {@inheritdoc}
    */
+  public static function buildResponse($view_id, $display_id, array $args = []) {
+    $response = parent::buildResponse($view_id, $display_id, $args);
+
+    // Add the content disposition header if a custom filename has been used.
+    // @todo Is there a way to retrieve an already-executed view?
+    $view = View::load($view_id)->getExecutable();
+    $view->setDisplay($display_id);
+    /** @var \Drupal\views\Plugin\views\display\DisplayPluginInterface $display */
+    $display = $view->getDisplay();
+    if ($display->getOption('filename')) {
```

Comment:
All this seems unnecessary, given the view has already executed at this point. The function `views_get_current_view()` returns nothing here though.